server/server_test.go

```diff
@@ -516,3 +517,164 @@ type okHTTPHandler struct{}
+							LoadBalancer: &types.LoadBalancer{Method: "Wrr"},
+						},
+					},
+				}
+			},
+			wantStatusCode: http.StatusOK,
+		},
+		{
+			desc: "No Frontend",
+			dynamicConfig: func(testServerURL string) *types.Configuration {
+				return &types.Configuration{
+					Frontends: map[string]*types.Frontend{},
+					Backends:  map[string]*types.Backend{},
+				}
+			},
+			wantStatusCode: http.StatusNotFound,
+		},
+		{
+			desc: "Empty Backend LB-Drr",
+			dynamicConfig: func(testServerURL string) *types.Configuration {
+				return &types.Configuration{
+					Frontends: map[string]*types.Frontend{
+						"frontend": {
+							EntryPoints: []string{"http"},
+							Backend:     "backend",
+							Routes:      map[string]types.Route{requestPath: {Rule: "Path:" + requestPath}},
+						},
+					},
+					Backends: map[string]*types.Backend{
+						"backend": {
+							Servers:      map[string]types.Server{},
+							LoadBalancer: &types.LoadBalancer{Method: "Drr"},
+						},
+					},
+				}
+			},
+			wantStatusCode: http.StatusServiceUnavailable,
+		},
+		{
+			desc: "Empty Backend LB-Drr Sticky",
+			dynamicConfig: func(testServerURL string) *types.Configuration {
+				return &types.Configuration{
+					Frontends: map[string]*types.Frontend{
+						"frontend": {
+							EntryPoints: []string{"http"},
+							Backend:     "backend",
+							Routes:      map[string]types.Route{requestPath: {Rule: "Path:" + requestPath}},
+						},
+					},
+					Backends: map[string]*types.Backend{
+						"backend": {
+							Servers:      map[string]types.Server{},
+							LoadBalancer: &types.LoadBalancer{Method: "Drr", Sticky: true},
+						},
+					},
+				}
+			},
+			wantStatusCode: http.StatusServiceUnavailable,
+		},
+		{
+			desc: "Empty Backend LB-Wrr",
+			dynamicConfig: func(testServerURL string) *types.Configuration {
+				return &types.Configuration{
+					Frontends: map[string]*types.Frontend{
+						"frontend": {
+							EntryPoints: []string{"http"},
+							Backend:     "backend",
+							Routes:      map[string]types.Route{requestPath: {Rule: "Path:" + requestPath}},
+						},
+					},
+					Backends: map[string]*types.Backend{
+						"backend": {
+							Servers:      map[string]types.Server{},
+							LoadBalancer: &types.LoadBalancer{Method: "Wrr"},
+						},
+					},
+				}
+			},
+			wantStatusCode: http.StatusServiceUnavailable,
+		},
+		{
+			desc: "Empty Backend LB-Wrr Sticky",
+			dynamicConfig: func(testServerURL string) *types.Configuration {
+				return &types.Configuration{
+					Frontends: map[string]*types.Frontend{
+						"frontend": {
+							EntryPoints: []string{"http"},
+							Backend:     "backend",
+							Routes:      map[string]types.Route{requestPath: {Rule: "Path:" + requestPath}},
+						},
+					},
+					Backends: map[string]*types.Backend{
+						"backend": {
+							Servers:      map[string]types.Server{},
+							LoadBalancer: &types.LoadBalancer{Method: "Wrr", Sticky: true},
+						},
+					},
+				}
+			},
```

Comment:
Thanks for the nice article recommendation. I applied the pattern there and I think the readability of the tests improved quite a lot. Furthermore I tried to keep the builder generic, so that they can easily be reused and extended by other tests in the future.